src/main/java/fact/io/hdureader/Fits.java

```diff
@@ -0,0 +1,194 @@
+ * HDUs can be accessed by their name.
+ *
+ *    Fits f = Fits.fromPath(p)
+ *    HDU events = f.getHDU("ZDrsCellOffsets");
+ *    BinTable binTable = events.getBinTable();
+ *
+ * BinTables can also directly be accessed by name if they exist.
+ *
+ *    Fits.fromPath(p).getBinTableByName("Events")
+ *                    .ifPresent(binTable -> {
+ *                        //do something with bintable
+ *                    })
+ *
+ * Data from the BinTable can be read using the BinTableReader and ZFitsHeapReader classes.
+ *
+ *    Fits f = new Fits(u);
+ *    BinTable events = f.getBinTableByName("Events").orElseThrow(IOException::new);
+ *
+ *    for(OptionalTypesMap<String, Serializable> p : BinTableReader.forBinTable(events)){
+ *      assertTrue(p.containsKey("Data"));
+ *    }
+ *
+ *
+ * Created by mackaiver on 03/11/16.
+ */
+public class Fits {
+    private static Logger log = LoggerFactory.getLogger(Fits.class);
+
+
+    private final URL url;
+    private final List<HDU>  hdus = new ArrayList<>();
+
+    private Map<String, HDU> hduNames = new HashMap<>();
+
+    public HDU primaryHDU;
+
+    /**
+     * Creates a fits instance from a path object without throwing checked exceptions immediately.
+     * @param path The path to the fits file
+     * @return a Fits object
+     */
+    public static Fits fromPath(Path path)  {
+        try {
+            return new Fits(path.toUri().toURL());
+        } catch (IOException e) {
+            throw new RuntimeException();
+        }
+    }
+
+    /**
+     * Creates a fits instance from a File object without throwing checked exceptions immediately.
+     * @param file the fits file
+     * @return a Fits object
+     */
+    public static Fits fromFile(File file)  {
+        if (!file.canRead()){
+            throw new RuntimeException("File " + file.toString() +  " is not readable");
+        }
+        try {
+            return new Fits(file.toURI().toURL());
+        } catch (IOException e) {
+            throw new RuntimeException();
+        }
+    }
+
+    public Fits(URL  url) throws IOException {
+        this.url = url;
+        DataInputStream stream = getDecompressedDataStream(url);
+        long absoluteHduOffsetInFile = 0;
+
+        try {
+            while (true) {
+                HDU h = new HDU(stream, url, absoluteHduOffsetInFile);
+                absoluteHduOffsetInFile += h.header.headerSizeInBytes + h.offsetToNextHDU();
+
+                hdus.add(h);
+
+                h.header.get("EXTNAME").ifPresent(name -> {
+                    hduNames.put(name, h);
+                });
+
+
+                ByteStreams.skipFully(stream, h.offsetToNextHDU());
+
+            }
+
+        } catch (EOFException e){
+            primaryHDU = hdus.get(0);
+            stream.close();
+
+            log.debug("A total of {} HDUs were found in the file.", hdus.size());
+        }
+    }
+
+    static DataInputStream getDecompressedDataStream(URL url) throws IOException {
```

Comment:
this name `getDecompressedDataStream` when I first read it somewhere further up ... I was wondering how you were able to decompresse the zfits data already that early in the process. Now I see that `decompress` refers to gzip only. Maybe renaming the function in a way, that makes clear this is about gzip only and not about our own zfits stuff might make sense.

I have no idea for a good name though.